tests/integration/monitoring/test_monitoring.py

```diff
@@ -115,3 +115,35 @@ def test_document_processed_total(port_generator):
             f'jina_document_processed_total{{endpoint="/foo",executor="DummyExecutor"}} 4.0'  # check that we nothing change on bar count
             in str(resp.content)
         )
+
+
+def test_disable_monitoring_on_pods(port_generator):
+    port0 = port_generator()
+    port1 = port_generator()
+
+    with Flow(monitoring=True, port_monitoring=port0).add(
```

Comment:
this should be a unit tests and not an integration one. U can check when the Flow construction happens that the `args` are what u expect. I am not saying to erase this test but to also keep the simpler one

Reply:
I added unit tests